src/hoc/requiresAuth.js

```diff
@@ -0,0 +1,56 @@
+//@ts-check
+import { navigate } from "@reach/router"
+import React, { useContext, useEffect, Fragment } from "react"
+import { UserContext } from "../../state"
+
+const RequiresAuth = ({ children }) => {
+  const userContext = useContext(UserContext)
+
+  // we use useEffect here to execute this logic only when the component mounts,
+  // this is to prevent ssr since localStorage is browser API
+  useEffect(() => {
+    ensureAuth()
+  }, [])
```

Comment:
if this is on component on mount only, then remove dependency array
```suggestion
  useEffect(() => {
    ensureAuth()
  })
```